base_layer/wallet/src/output_manager_service/master_key_manager.rs

```diff
@@ -0,0 +1,222 @@
+            key_manager_state.master_key.clone(),
+            KEY_MANAGER_COINBASE_SCRIPT_BRANCH_KEY.to_string(),
+            0,
+        );
+
+        let rewind_key_manager = KeyManager::<PrivateKey, KeyDigest>::from(
+            key_manager_state.master_key.clone(),
+            KEY_MANAGER_RECOVERY_VIEWONLY_BRANCH_KEY.to_string(),
+            0,
+        );
+        let rewind_key = rewind_key_manager.derive_key(0)?.k;
+
+        let rewind_blinding_key_manager = KeyManager::<PrivateKey, KeyDigest>::from(
+            key_manager_state.master_key,
+            KEY_MANAGER_RECOVERY_BLINDING_BRANCH_KEY.to_string(),
+            0,
+        );
+        let rewind_blinding_key = rewind_blinding_key_manager.derive_key(0)?.k;
+
+        let rewind_data = RewindData {
+            rewind_key,
+            rewind_blinding_key,
+            proof_message: [0u8; REWIND_USER_MESSAGE_LENGTH],
+        };
+
+        Ok(Self {
+            utxo_key_manager: Mutex::new(utxo_key_manager),
+            utxo_script_key_manager: Mutex::new(utxo_script_key_manager),
+            coinbase_key_manager: Mutex::new(coinbase_key_manager),
+            coinbase_script_key_manager: Mutex::new(coinbase_script_key_manager),
+            rewind_data,
+            db,
+        })
+    }
+
+    pub fn rewind_data(&self) -> &RewindData {
+        &self.rewind_data
+    }
+
+    /// Return the next pair of (spending_key, script_private_key) from the key managers. These will always be generated
+    /// in tandem and at corresponding increments
+    pub async fn get_next_spend_and_script_key(&self) -> Result<(PrivateKey, PrivateKey), OutputManagerError> {
+        let mut km = self.utxo_key_manager.lock().await;
+        let key = km.next_key()?;
+
+        let mut skm = self.utxo_script_key_manager.lock().await;
+        let script_key = skm.next_key()?;
+
+        self.db.increment_key_index().await?;
+        Ok((key.k, script_key.k))
+    }
+
+    pub async fn get_script_key_at_index(&self, index: u64) -> Result<PrivateKey, OutputManagerError> {
+        let skm = self.utxo_script_key_manager.lock().await;
+        let script_key = skm.derive_key(index)?;
+        Ok(script_key.k)
+    }
+
+    pub async fn get_coinbase_spend_and_script_key_for_height(
+        &self,
+        height: u64,
+    ) -> Result<(PrivateKey, PrivateKey), OutputManagerError> {
+        let km = self.coinbase_key_manager.lock().await;
+        let spending_key = km.derive_key(height)?;
+
+        let mut skm = self.coinbase_script_key_manager.lock().await;
+        let script_key = skm.next_key()?;
+        Ok((spending_key.k, script_key.k))
+    }
+
+    /// Return the Seed words for the current Master Key set in the Key Manager
+    pub async fn get_seed_words(&self) -> Result<Vec<String>, OutputManagerError> {
+        Ok(from_secret_key(
+            self.utxo_key_manager.lock().await.master_key(),
+            &MnemonicLanguage::English,
```

Comment:
nit: Should this not rather be a config option for `MasterKeyManager`. 
This seems like a smell, hard-coded option within a function. 
I would rather say let `MasterKeyManager` have a field language and hard code it in there for now.

Reply:
Fair enough, gave this function a argument to specify the language and added it to the Output Manager Config and its a default now.